### PR TITLE
rgw: add PUT and POST req support to data cache

### PR DESCRIPTION
**Pull request by mkogan1**
facilitates the full usage of the Nginx cache endpoint
with s3 tools that support AWSv4
like s3cmd,aws-cli, benchmarking tools like hsbench
and also hadoop/s3a.

Co-authored-by: Or Friedmann <ofriedma@redhat.com>
Signed-off-by: Mark Kogan <mkogan@redhat.com>

## Checklist
- [ ] References tracker ticket
- [x] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by mkogan1** [comment]
jenkins render docs

**Comment by mkogan1** [comment]
jenkins render docs

File: doc/radosgw/rgw-cache.rst [hunk 1]
```diff
@@ -86,42 +90,62 @@ Put in-place your nginx configuration files and edit them according to your envi
...
 
 Run nginx(openresty):
+
 ::
 
 $ sudo systemctl restart nginx
+
+Appendix
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+**A note about performance:** In certain instances like development environment, disabling the authontication by commenting the following line in `nginx-default.conf`:
```
Comment (mattbenjamin) [inline doc/radosgw/rgw-cache.rst:144]: sp. authentication

File: src/rgw/rgw_auth_s3.cc [hunk 2]
```diff
@@ -571,7 +571,7 @@ get_v4_canonical_headers(const req_info& info,
     }
     const char* const t = info.env->get(token_env.c_str());
     if (!t) {
-      dout(10) << "warning env var not available" << dendl;
+      dout(10) << "warning env var not available " << token_env.c_str() << dendl;
```
Comment (mattbenjamin) [inline src/rgw/rgw_auth_s3.cc:574]: what does token_env contain here?  will this make sense when printed?

Comment (ofriedma) [reply src/rgw/rgw_auth_s3.cc:574]: It contains the header name that has been in the signedheaders field but the RGW didn't get it

File: doc/radosgw/rgw-cache.rst (hunk 1 above)
Comment (mkogan1) [reply doc/radosgw/rgw-cache.rst]: thank you, pushed a fix for spelling issues.

**Comment by mkogan1** [comment]
jenkins render docs

**Pull request merged**